internal/middleware/compression.go

```diff
@@ -142,6 +172,15 @@ func newBodyReader(r *http.Request) (io.ReadCloser, error) {
 			return nil, err
 		}
 		return zr, nil
+	case "snappy":
```

Comment:
The behaviour of `snappy` and `zstd` are different than `gzip` and `zlib`.
While `gzip` and `zlib` return error in `newBodyReader()` when uncompressed text comes, `snappy` and `zstd` don't return error explicitly.

Reply:
I don't see a good way around this right now.  It might be best to simply not add decompression support for snappy and zstd in this PR.  We can open a new issue for each of them to track the desire to have the support.